kong/globalpatches.lua

```diff
@@ -589,6 +591,32 @@ return function(options)
       return sock
     end
 
+    -- OTel-formatted logs feature
+    local dynamic_hook = require "kong.dynamic_hook"
+    local log_called = false
+    _G.ngx.log = function(...)
+      if log_called then
+        -- avoid recursive loops
+        -- relies on the patch to NOT yield
+        old_ngx_log(ngx.ERR, debug.traceback("concurrent execution detected for: ngx.log", 2))
```

Comment:
``` suggestion
        old_ngx_log(ngx.ERR, debug.traceback("concurrent execution detected for: ngx.log", 2))
        log_called = false
```

Reply:
not sure about this. If we end up here and `log_called` is still **true** it means something is still executing the patched `ngx.log` function (which yielded for some reason). In that case, we probably want to leave the flag to true so that it remains in a failing state if executed again. WDYT?

Reply:
There is another case.

Let's say the `log_level` is invalid and the `old_ngx_log` [raises a Lua error](https://github.com/openresty/lua-nginx-module/blob/6738c3a3b0d62b98ff35acf7f7c4a191207da918/src/ngx_http_lua_log.c#L54) to terminate the entire process flow of this request.

Consequently, the `log_called` will be `true` until we restart this worker process, and any following log lines cannot be sent to the otlp server.

Reply:
Looks like whether we set the `log_called = false` or not here, there is a critical bug in this detecting logic?

If we do it, we might still run into infinity recursive. If not, we might have to restart the worker process to resume the Otel log shipping.

Am I right? Not sure about it.

Reply:
exactly @ADD-SP you are right and I think it's preferrable for the second scenario to happen, i.e. for OTel log to stop working if we end up in such a failing state (which is unexpected after all), and avoid infinite recursion at all cost. Let me know if you agree.

Reply:
>  for OTel log to stop working if we end up in such a failing state

How to detect this failing state? The entire code flow has been terminated, how to get such knowledge?

Reply:
(conversation moved offline, I'll update the thread with our conclusions)

Reply:
as agreed offline, moving the old_ngx_log call up should address these concerns, I made the change.